app/clients/file_client.rb

```diff
@@ -13,7 +13,7 @@ class FileClient
   # - the contents of the file.
 
   def self.save(user_id, folder, filename, content)
-    filepath = self.filepath user_id, folder, filename
+    filepath = filepath user_id, folder, filename
```

Comment:
Got it. Private methods in ruby cannot be called with an explicit receiver. This is true both for instance and class methods. That's why ```self.filepath``` fails, ```FileClient.filepath``` would fail as well. Private methods must be called without an explicit receiver, using the implicit ```self``` receiver. It seems the only option here is to call ```filepath``` the way you were doing; this means it would be a good idea to change the name of the variable. I'll have a look at it tomorrow, this failure is happening across several classes now.

Another weird quirk of ruby: ```self.foo``` fails, but calling ```foo``` is ok, even though it implictly uses ```self``` as the receiver. 

I swear I thought I had a good grasp of ruby, but this basic misunderstunding is rather embarrasing.